tests/semantic_analyser.cpp

```diff
@@ -974,9 +965,9 @@ kprobe:f { @a[1] = 1; delete(@a, @a); }
   test(R"(kprobe:f { @y[1, "longerstr"] = 5; delete(@y[1, "hi"]); })");
 
   test_error("kprobe:f { @x = 1; @y = 5; delete(@x, @y); }", R"(
-stdin:1:39-41: ERROR: Argument mismatch for @x: trying to access with arguments: 'int64' when map expects no arguments
+stdin:1:28-37: ERROR: delete() expects a map with explicit keys (non-scalar map)
```

Comment:
Again, this should read something like "delete(), when called with 2 arguments, expects a map with explicit keys (non-scalar map) as it's first argument"

Reply:
Can I get away with something like "call to delete() expects ..." implying that it is just this call? Otherwise everything gets special cased around `delete` vs. all these other map calls (e.g. `has_key`, `len`, etc.).

Reply:
So what would be the final error message?

Reply:
```
stdin:1:28-37: ERROR: call to delete() expects a map with explicit keys (non-scalar map)
kprobe:f { @x = 1; @y = 5; delete(@x, @y); }
                           ~~~~~~~~~
```